sycl/test-e2e/Config/select_device.cpp

```diff
@@ -456,510 +386,274 @@ int main() {
             addEscapeSymbolToSpecialCharacters(name);
             std::string ver = dev.get_info<info::device::driver_version>();
             size_t pos = 0;
-            if (((plt.get_backend() == backend::opencl) &&
-                 (sycl_be.find("opencl") != std::string::npos)) ||
-                ((plt.get_backend() == backend::ext_oneapi_level_zero) &&
-                 (sycl_be.find("level_zero") != std::string::npos))) {
+            if ((plt.get_backend() == backend::opencl) ||
+                (plt.get_backend() == backend::ext_oneapi_level_zero)) {
               if ((pos = ver.find(".")) == std::string::npos) {
                 throw std::runtime_error("Malformed syntax in version string");
               }
               pos++;
               size_t start = pos;
+              // FIXME: What is going on here?!! It's insanity to expect a
+              // different result than few lines above. Is that a weirdest
+              // possible way to write "pos -= 1;"???
```

Comment:
This comment is new.

Reply:
It's my second  favorite comment in the code base, my most being  " // Drunk, fix later. "